### PR TITLE
feat: weights

### PR DESCRIPTION
**Pull request by edisontim**

### CODE REVIEW

File: client/src/dojo/modelManager/ArmyMovementManager.ts [hunk 1]
```diff
@@ -411,14 +408,34 @@ export class ArmyMovementManager {
...
-    const capacity = armyCapacity?.weight_gram || 0n;
-    const weight = (armyWeight?.value || 0n) / BigInt(EternumGlobalConfig.resources.resourcePrecision);
+    if (!armyEntity || !armyCapacity || !armyWeight) return 0n;
 
-    return capacity * troopQty - weight;
+    return getRemainingCapacity(armyEntity, armyCapacity, armyWeight!);
   };
 }
+
+const getRemainingCapacity = (
```
[the remark this answers is not in the document]
Comment (aymericdelab) [reply client/src/dojo/modelManager/ArmyMovementManager.ts:417]: (we should move TravelPaths from it as well imo)

File: client/src/hooks/helpers/useArmies.tsx [hunk 2]
```diff
@@ -73,57 +73,41 @@ const formatArmies = (
...
-      let capacityClone = structuredClone(capacity);
-      if (capacityClone) {
-        capacityClone.weight_gram = capacityClone.weight_gram * quantityClone.value;
-      } else {
-        capacityClone = {
-          entity_id: army.entity_id,
-          weight_gram: BigInt(EternumGlobalConfig.carryCapacity.army) * quantityClone.value,
-        };
-      }
+      const capacity = structuredClone(getComponentValue(Capacity, armyEntityId));
```
Comment (aymericdelab) [inline client/src/hooks/helpers/useArmies.tsx:104]: is there a need for structuredClone here ?

File: client/src/hooks/helpers/useArmies.tsx [hunk 3]
```diff
@@ -73,57 +73,41 @@ const formatArmies = (
...
-      let weightClone = structuredClone(weight);
-      if (weightClone) {
-        weightClone.value = weightClone.value / BigInt(EternumGlobalConfig.resources.resourcePrecision);
-      } else {
-        weightClone = {
-          entity_id: army.entity_id,
-          value: 0n,
-        };
-      }
+      const weightComponentValue = structuredClone(getComponentValue(Weight, armyEntityId));
```
Comment (aymericdelab) [inline client/src/hooks/helpers/useArmies.tsx:107]: is there a need for structured clone here ?

File: client/src/ui/modules/navigation/TopMiddleNavigation.tsx [hunk 4]
```diff
@@ -54,8 +54,7 @@ export const TopMiddleNavigation = () => {
   const setPreviewBuilding = useUIStore((state) => state.setPreviewBuilding);
   const selectedQuest = useQuestStore((state) => state.selectedQuest);
 
-  const selectedHex = useUIStore((state) => state.selectedHex);
-
+  console.log("realmEntityId", realmEntityId);
```
Comment (aymericdelab) [inline client/src/ui/modules/navigation/TopMiddleNavigation.tsx:57]: remove console.log

File: config/index.ts [hunk 5]
```diff
@@ -37,16 +37,16 @@ const nodeUrl = process.env.VITE_PUBLIC_DEV === "true" ? "http://127.0.0.1:5050/
...
-await setWeightConfig(account, provider);
-await setCombatConfig(account, provider);
-await setCapacityConfig(account, provider);
-await setSpeedConfig(account, provider);
+// await setProductionConfig(account, provider);
+// await setBuildingCategoryPopConfig(account, provider);
+// await setPopulationConfig(account, provider);
+// await setBuildingConfig(account, provider);
+// await setResourceBuildingConfig(account, provider);
+// await setWeightConfig(account, provider);
```
Comment (aymericdelab) [inline config/index.ts:45]: don't forget to uncomment

File: sdk/packages/eternum/src/constants/global.ts [hunk 6]
```diff
@@ -43,11 +43,11 @@ export const EternumGlobalConfig = {
   },
   tick: {
     defaultTickIntervalInSeconds: 1,
-    armiesTickIntervalInSeconds: 7200, // 2hrs
+    armiesTickIntervalInSeconds: 30, // 2hrs
```
Comment (aymericdelab) [inline sdk/packages/eternum/src/constants/global.ts:46]: remove

File: sdk/packages/eternum/src/config/index.ts [hunk 7]
```diff
@@ -213,7 +213,7 @@ export const setCapacityConfig = async (account: Account, provider: EternumProvi
   const txDonkey = await provider.set_capacity_config({
     signer: account,
     entity_type: DONKEY_ENTITY_TYPE,
-    weight_gram: EternumGlobalConfig.carryCapacity.donkey * EternumGlobalConfig.resources.resourcePrecision,
+    weight_gram: EternumGlobalConfig.carryCapacity.donkey,
```
Comment (aymericdelab) [inline sdk/packages/eternum/src/config/index.ts:216]: call it carrCapacityGrams (same for weight) in EternumGlobalConfig

File: sdk/packages/eternum/src/constants/global.ts [hunk 8]
```diff
@@ -43,11 +43,11 @@ export const EternumGlobalConfig = {
...
   },
   carryCapacity: {
```
Comment (aymericdelab) [inline sdk/packages/eternum/src/constants/global.ts:48]: add comment that says gram per unit (without resource precision)

File: sdk/packages/eternum/src/constants/resources.ts [hunk 9]
```diff
@@ -415,36 +415,36 @@ export const getLevelingCost = (newLevel: number): { resourceId: number; amount:
 export const WEIGHTS: {
```
Comment (aymericdelab) [inline sdk/packages/eternum/src/constants/resources.ts:415]: WEIGHTS_IN_GRAMS

Comment (aymericdelab) [reply sdk/packages/eternum/src/constants/resources.ts:415]: or add comment

File: client/src/hooks/helpers/useArmies.tsx (hunk 2 above)
Comment (edisontim) [reply client/src/hooks/helpers/useArmies.tsx:104]: removed it, my bad

**Review by aymericdelab — APPROVED** [review]

**Pull request merged**